cucumber-expressions/java/src/main/java/io/cucumber/cucumberexpressions/GroupBuilder.java

```diff
@@ -1,14 +1,25 @@
 package io.cucumber.cucumberexpressions;
 
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.Iterator;
 import java.util.List;
 import java.util.regex.Matcher;
 
-final class GroupBuilder {
+import org.apiguardian.api.API;
+
+@API(status = API.Status.EXPERIMENTAL)
+public final class GroupBuilder {
```

Comment:
Semantically when exposed this wouldn't be a builder but just a `Group`.

Reply:
The problem is that there is already a "Group" class, the GroupBuilder builds Group(s) out of the Matched groups of a RegExp. I can rename thsi to TreeGroup if you think thats suits better?

Reply:
Or do you think the (internal) GroupBuilder should be converted to public Groups instances as they have very similar API?